python/ray/data/dataset.py

```diff
@@ -451,6 +452,62 @@ def get_dataset_id(self) -> str:
         """
         return self._plan.get_dataset_id()
 
+    @PublicAPI(api_group=BT_API_GROUP)
+    def load_from_uris(
+        self,
+        uri_column: str,
+        *,
+        decode_fn: Optional[Callable[[io.RawIOBase], Iterator[DataBatch]]] = None,
+    ):
+        """Load binary data from a column of URIs.
+
+        Args:
+            uri_column: The name of the column containing the URIs.
+            decode_fn: A function that takes a file-like object and returns an iterator
+                of data batches.
+
+        Returns:
+            A new Dataset with the decoded data.
+        """
+        RANDOM_SAMPLE_SIZE = 10
+        import math
+        import random
+
+        if decode_fn is None:
+
+            def default_decode_fn(f: io.RawIOBase) -> Iterator[DataBatch]:
+                return f
+
+            decode_fn = default_decode_fn
+
+        # Step 1: Take all paths for sampling and to determine how many there are.
+        paths = self.select_columns([uri_column]).take_all()
+
+        # Step 2: Sample paths to estimate total in memory size after reading.
+        sampled_paths = [
+            row[uri_column] for row in random.sample(paths, RANDOM_SAMPLE_SIZE)
+        ]
+        total_sampled_size = (
+            ray.data.read_binary_files(sampled_paths)
+            .map(lambda r: {"size_bytes": len(r["bytes"])})
+            .sum("size_bytes")
+        )
```

Comment:
Can we use `FileMetadataProvider` to get the file sizes? It'll be much cheaper than actually reading the data

Reply:
And actually, if we're just listing all the file sizes with `FileMetadataProvider`, I think we can just take the mean.